Update index.md (#908)
root-project/web: for_developers/run_the_tests/index.md

  nav: "for_developers"
---

Running all ROOT tests on at least a platform before sending a PR or before pushing anything to the ROOT repository is **mandatory**.
The ROOT continuous integration infrastructure allows to automatically check the PR also on a battery of platform, for example including macos and linux boxes.
In order to be able to run all tests, the simple steps below need to be followed

> ## Prepare Your Development Area for Testing and Building ROOT

Reviewer: ```suggestion
## Prepare Your Development Area for Building and Testing ROOT
```

Create a directory to store the local repositories. Within the directory, clone both the root project as well as the [roottest](https://github.com/root-project/roottest) test suite.
```
mkdir RootDevelopment
cd RootDevelopment
git clone https://github.com/root-project/root
git clone https://github.com/root-project/roottest
```

> Create a build folder, run the cmake command to build out rootest as a part of root by enabling the 'testing' and 'roottest' option:

Reviewer: ```suggestion
Create a build folder, run the cmake command to configure and build root and rootest:
```

```
mkdir build
cd build
echo Note the cmake options
cmake -DCMAKE_BUILD_TYPE=Debug -Dtesting=ON -Droottest=ON ../root
cmake --build . -j8
```

## Inspect and run the tests
Tests can then be executed using the ctest command in the build directory:

```

> ctest -N                (list all tests)

Reviewer: Instead of listing the options here, can we just say "run `ctest --help` to see available options"?

Author: Yes, this would help condense it further. But we ought to still mention the basic "Run all tests" command to get users up and running with the test suite quickly.

  nav: "for_developers"
---

Running all ROOT tests on at least a platform before sending a PR or before pushing anything to the ROOT repository is **mandatory**.
The ROOT continuous integration infrastructure allows to automatically check the PR also on a battery of platform, for example including macos and linux boxes.
In order to be able to run all tests, the simple steps below need to be followed

## Prepare Your Development Area for Testing and Building ROOT

> #### Option 1: Build roottest as part of a ROOT build

Reviewer: I would prefer that we don't offer options: we should keep at as simple as possible. And we should use as few words as possible: people don't come here to be entertained :-) Can you adjust this sightly, picking the option you prefer?

Author: Sure, will keep option 1: build roottest along with ROOT build
